pkgs/flox/default.nix

```diff
@@ -43,6 +45,14 @@
       FLOX_GH_BIN = "${flox-gh}/bin/flox-gh";
       GH_BIN = "${gh}/bin/gh";
       FLOX_SH_PATH = flox-bash.outPath;
+      ENV_FROM_LOCKFILE_PATH = ../../assets/mkEnv/env-from-lockfile.nix;
+      BUILD_ENV_BIN = writeScript "build-env" ''
+        NIX="$1";
+        SYSTEM="$2";
+        LOCKFILE_PATH="$3";
+        OUTLINK_PREFIX="$4";
+        "$NIX" build --file ${ENV_FROM_LOCKFILE_PATH} --out-link "$OUTLINK_PREFIX" --print-out-paths --arg lockfilePath "$LOCKFILE_PATH" --argstr system "$SYSTEM"
```

Comment:
question nonblocking: what's the point of having a script do this instead of just running nix build from Rust?

Reply:
It's important to be able to build an environment separately from activating the environment for testing purposes. It's easier to do this with a separate executable. Looking ahead, we'll be building another executable that implements our own `buildEnv`. At that point it's just a matter of switching out which executable we're calling.

Reply:
(nonblocking) even when we write `ourBuildEnv` won't it be inside the Nix build? I.e. replace `linkFarmFromDrvs` with `ourBuildEnv` in `ENV_FROM_LOCKFILE_PATH`? But then we'll still need to run a one off `nix build` command somewhere, and I don't get why we'd want a script that only does that. As far as testing can't we just write a unit test?

Reply:
Making it a separate executable enables future SaaS work to build environments without the full CLI.
It also makes it far simpler to create a drop in replacement in the future.

Reply:
(nonblocking) I agree with @mkenigs on that this looks kinda weird and we can just as well use runix to run the command. expose as a separate command for floxEM etc..
But i accept an armadillo on this 

<details>
<summary>Details</summary>

![image](https://github.com/flox/flox/assets/7040031/e86bf547-d904-4808-be9e-efc6794815a5)


</details>

Reply:
- I don't want this exposed as a separate command for floxEM to use, that's how we ended up with 4 million subcommands. 
- Just imagine I replace this script with a compiled binary. Basically nothing changes.
- `even when we write ourBuildEnv won't it be inside the Nix build?` -> I don't think so, I think the plan is for the environment builder to be a separate binary that does the entire build itself.

I'm resolving this conversation because this code will live for < 8 weeks and this gets us to a demo.